File: backend/app/schemas/issue.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import IssuePriority, IssueStatus


class IssueBase(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    description: str | None = None
    status: IssueStatus = IssueStatus.TODO
    priority: IssuePriority = IssuePriority.MEDIUM
    assignee_id: uuid.UUID | None = None


class IssueCreate(IssueBase):
    # No authentication yet, so the creator is supplied explicitly.
    created_by_id: uuid.UUID


class IssueUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    status: IssueStatus | None = None
    priority: IssuePriority | None = None
    assignee_id: uuid.UUID | None = None


class IssueResponse(IssueBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    created_by_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/issue.ts
export type IssueStatus = 'todo' | 'in_progress' | 'done'
export type IssuePriority = 'low' | 'medium' | 'high' | 'critical'

export interface Issue {
  id: string
  project_id: string
  title: string
  description: string | null
  status: IssueStatus
  priority: IssuePriority
  assignee_id: string | null
  created_by_id: string
  created_at: string
  updated_at: string
}

/** Shape produced by the create/edit issue form; the API layer narrows this
 * to Partial<IssueFormValues> for updates. */
export interface IssueFormValues {
  title: string
  description: string | null
  status: IssueStatus
  priority: IssuePriority
  assignee_id: string | null
}
